### PR TITLE
New error improvements for 1.7 acceleration

### PR DESCRIPTION
**Pull request by KyleDavisSA**
This PR improves the error messages for the acceleration schemes, and supersedes PR https://github.com/precice/precice/pull/729. This PR also includes the review from PR https://github.com/precice/precice/pull/729 and will be closed.

### CODE REVIEW

File: src/acceleration/BaseQNAcceleration.cpp [hunk 1]
```diff
@@ -212,7 +214,9 @@ void BaseQNAcceleration::updateDifferenceMatrices(
 
       if (2 * getLSSystemCols() >= getLSSystemRows())
         PRECICE_WARN(
-            "The number of columns in the least squares system exceeded half the number of unknowns at the interface. The system will probably become bad or ill-conditioned and the quasi-Newton acceleration may not converge. Maybe the number of allowed columns (maxIterationsUsed) should be limited.");
+            "The number of columns in the least squares system exceeded half the number of unknowns at the interface. "
+            << "The system will probably become bad or ill-conditioned and the quasi-Newton acceleration may not "
+            << "converge. Maybe the number of allowed columns (maxIterationsUsed) should be limited.");
```
Comment (uekerman) [inline src/acceleration/BaseQNAcceleration.cpp:219]:
```suggestion
            << "converge. Maybe the number of allowed columns ("max-used-iterations") should be limited.");
```

File: src/acceleration/BroydenAcceleration.cpp [hunk 2]
```diff
@@ -75,7 +75,7 @@ void BroydenAcceleration::computeQNUpdate(Acceleration::DataMap &cplData, Eigen:
 
   PRECICE_DEBUG("currentColumns=" << _currentColumns);
   if (_currentColumns > 1) {
-    PRECICE_ERROR("truncated IMVJ no longer supported, needs to be parallelized and datastructures need to be changed to Eigen datastructures.");
+    PRECICE_ERROR("Truncated IMVJ no longer supported. Use IMVJ with restart mode.");
```
Comment (uekerman) [inline src/acceleration/BroydenAcceleration.cpp:78]:
```suggestion
    PRECICE_ERROR("Truncated IMVJ is no longer supported. Please use IMVJ with restart mode instead.");
```

File: src/acceleration/BaseQNAcceleration.cpp [hunk 3]
```diff
@@ -216,7 +216,7 @@ void BaseQNAcceleration::updateDifferenceMatrices(
         PRECICE_WARN(
             "The number of columns in the least squares system exceeded half the number of unknowns at the interface. "
             << "The system will probably become bad or ill-conditioned and the quasi-Newton acceleration may not "
-            << "converge. Maybe the number of allowed columns (maxIterationsUsed) should be limited.");
+            << "converge. Maybe the number of allowed columns ("max-used-iterations") should be limited.");
```
Comment (uekerman) [inline src/acceleration/BaseQNAcceleration.cpp:219]:
```suggestion
            << "converge. Maybe the number of allowed columns (\"max-used-iterations\") should be limited.");
```

**Comment by fsimonis** [comment]
Is this PR ready to be merged?

**Pull request merged**